### PR TITLE
Bug fix in ExtraCUDACopy and remove unstable lints for release

### PR DESCRIPTION
**Pull request by davidchencsl**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #82693
* #82261

### CODE REVIEW

File: torch/profiler/_pattern_matcher.py [hunk 1]
```diff
@@ -609,8 +628,8 @@ def report_all_anti_patterns(prof,
     report_dict: Dict = {}
     anti_patterns = [
         ExtraCUDACopyPattern(prof, should_benchmark),
-        ForLoopIndexingPattern(prof, should_benchmark),
-        FP32MatMulPattern(prof, should_benchmark),
+        # ForLoopIndexingPattern(prof, should_benchmark),
```
Comment (robieta) [inline torch/profiler/_pattern_matcher.py:631]: These are still tested, right?

**Review by robieta — APPROVED** [review]

File: torch/profiler/_pattern_matcher.py (hunk 1 above)
Comment (davidchencsl) [reply torch/profiler/_pattern_matcher.py:631]: Yes, the test entry is not report_anti_patterns, but rather the pattern classes :)

File: torch/profiler/_pattern_matcher.py [hunk 2]
```diff
@@ -176,7 +178,24 @@ def match(self, event):
...
+            return False
+        if not event.children:
+            return False
+        event = event.children[-1]
+        if event.name() != "aten::copy_":
+            return False
+        if not event.children:
+            return False
+        event = event.children[0]
+        if event.name() != "cudaMemcpyAsync":
```
Comment (jeffdaily) [inline torch/profiler/_pattern_matcher.py:195]: This is probably why rocm CI was failing.  Exact string matching on CUDA APIs.

Comment (davidchencsl) [reply torch/profiler/_pattern_matcher.py:195]: 👍 Thanks for pointing it out!

Comment (jeffdaily) [reply torch/profiler/_pattern_matcher.py:195]:
```suggestion
        if event.name() not in ("cudaMemcpyAsync", "hipMemcpyWithStream"):
```

If you rework your PR, the above should fix ROCm CI.

**Comment by jeffdaily** [comment]
I closed the issue that skipped the failing rocm ci, pending a rework of this PR.

https://github.com/pytorch/pytorch/issues/82781

**Pull request closed without merging**